src/Stoa.ts

```diff
@@ -23,14 +24,85 @@ class Stoa {
             }
         });
 
+        /**
+         * When a request is received through the `/validators` handler
+         * Validators based on the block height if there is a height.
+         * If height null the most up to date state is expected.
+         */
         this.stoa.get("/validators",
             (req: express.Request, res: express.Response, next: express.NextFunction) => {
-            res.send(req.query.height);
+
+                var height: number = Number(req.query.height);
+
+                this.ledger_storage.getValidatorsAPI(height, null, (err: Error | null, rows: any[]) =>
+                {
+                    if (err != null)
+                    {
+                        console.error("Failed to data lookup to the DB: " + err);
+                        res.status(500).send("Failed to data lookup");
+                        return;
+                    }
+
+                    if (rows.length)
+                    {
+                        let out_put:Array<ValidatorData> = new Array<ValidatorData>();
+
+                        for (const row of rows)
+                        {
+                            let preimage: IPreimage = {distance: row.distance, hash: '0'} as IPreimage;
+                            var validator: ValidatorData | null =
+                                new ValidatorData(row.address, row.enrolled_at, row.stake, preimage);
+                            out_put.push(validator);
+                        }
+                        res.status(200).send(JSON.stringify(out_put));
+                    }
+                    else
+                    {
+                        res.status(200).send("no data");
```

Comment:
Actually this should never happen. There must always be validators active at any block height.

Reply:
Yes, that's right. It won't happen, but if the height of the lookup condition exceeds the current block height, there may be no data. I will change.`res.status(200).send();`

Reply:
Then it should return some kind of status code so the client knows it called with invalid parameters. For example `204` like Mathias suggested: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204